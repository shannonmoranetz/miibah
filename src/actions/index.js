export const SET_AMIIBOS = 'SET_AMIIBOS';
export const ADD_TO_WISHLIST = 'ADD_TO_WISHLIST';
export const ADD_TO_COLLECTED = 'ADD_TO_COLLECTED';
export const GET_WISHLIST = 'GET_WISHLIST';
export const GET_COLLECTED = 'GET_COLLECTED';
export const SET_LOADING = 'SET_LOADING';
export const SET_ERROR = 'SET_ERROR';

export const setAmiibos = amiibos => ({
  type: SET_AMIIBOS,
  amiibos
});

export const addToWishlist = amiibo => ({
  type: ADD_TO_WISHLIST,
  amiibo
});

export const addToCollected = amiibo => ({
  type: ADD_TO_COLLECTED,
  amiibo
});

export const getWishlist = wishlist => ({
  type: GET_WISHLIST,
  wishlist
});

export const getCollected = collected => ({
  type: GET_COLLECTED,
  collected
});

export const setLoading = isLoading => ({
  type: SET_LOADING,
  isLoading
});

export const setError = error => ({
  type: SET_ERROR,
  error
});

