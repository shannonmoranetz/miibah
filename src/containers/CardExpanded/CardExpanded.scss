.CardExpanded {
  z-index: 1;
  position: absolute;
  top: 25%;
  right: 0;
  bottom: 0;
  left: 0;
}

.inner-card {
  margin: 0px auto;
  border: 6px solid $darkblue;
  display: inline-block;
  font-family: $body-font;
  width: 80%;
  color: white;
  background-color: $ice;
  display: flex;
  justify-content: space-around;
  align-items: center;
  text-transform: lowercase;
}

.card-info {
  margin-top: auto;
  margin-bottom: auto;
}

.card-name {
  font-family: $heading-font;
  font-size: 40px;
}

.image-expanded {
  margin: 30px auto;
  height: 300px;
}

.card-amiibo-series,
.card-game-series,
.card-date {
  font-size: 25px;
}

.exit-button {
  margin-bottom: 30%;
  text-align: center;
  padding: 5px 15px;
  font-size: 30px;
  border: 2px solid white;
  background-color: $salmon;
}