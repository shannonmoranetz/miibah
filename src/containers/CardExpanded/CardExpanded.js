import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addToWishlist, addToCollected }  from '../../actions';
import { Link } from 'react-router-dom';

export class CardExpanded extends Component {

  addAmiiboToWishlist = async () => {
    let amiibo = this.props;
    await this.props.addToWishlist(amiibo);
    localStorage.setItem('wishlist', JSON.stringify(this.props.wishlist));
  }

  addAmiiboToCollected = async () => {
    let amiibo = this.props;
    await this.props.addToCollected(amiibo);
    localStorage.setItem('collected', JSON.stringify(this.props.collected));
  }

  render() {
    return (
      <div className="CardExpanded">
        <div className="inner-card">
        <Link to="/" className="exit-button">x</Link>
          <div className="card-image-container">
            <img className="image-expanded" src={this.props.image} alt={`${this.props.name} amiibo`}/>
          </div>
          <div className="card-info">
            <h2 className="card-name">{this.props.name}</h2>
            <p className="card-amiibo-series"> amiibo series: {this.props.amiiboSeries}</p>
            <p className="card-game-series">game series: {this.props.gameSeries}</p>
            <p className="card-date">release date: {this.props.release.na}</p>
          </div>
          <div className="expanded-buttons">
            <button onClick={this.addAmiiboToWishlist} className="wish-button">wish</button>
            <button onClick={this.addAmiiboToCollected} className="collect-button">collect</button>
          </div>
      </div>
    </div>
    );
  }
}

export const mapStateToProps = state => ({
  wishlist: state.wishlist,
  collected: state.collected
});

export const mapDispatchToProps = dispatch => ({
  addToWishlist: (amiibo) => dispatch(addToWishlist(amiibo)),
  addToCollected: (amiibo) => dispatch(addToCollected(amiibo))
});

export default connect(mapStateToProps, mapDispatchToProps)(CardExpanded);
