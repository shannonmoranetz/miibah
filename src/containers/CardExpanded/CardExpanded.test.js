import React from 'react';
import { shallow } from 'enzyme';
import { CardExpanded, mapStateToProps, mapDispatchToProps } from '../CardExpanded/CardExpanded';
import { addToWishlist, addToCollected } from '../../actions';

const propsMock = {
  addAmiiboToWishlist: jest.fn(),
  addAmiiboToCollected: jest.fn(),
  addToWishlist: jest.fn(),
  addToCollected: jest.fn(),
  name: 'pikachu',
  amiiboSeries: 'pokemon',
  gameSeries: 'pokemon',
  image: 'pikachu.img',
  release: 2/20/2019,
  id: 1
};

describe('CardExpanded', () => {
  let wrapper; 
  beforeEach(() => {
    wrapper = shallow(<CardExpanded {...propsMock} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addToWishlist with an amiibo param when addAmiiboToWishlist is called', async () => {
    wrapper.instance().addAmiiboToWishlist();
    expect(propsMock.addToWishlist).toHaveBeenCalledWith(propsMock);
  })

  it('should call addToCollected with an amiibo param when addAmiiboToCollected is called', async () => {
    wrapper.instance().addAmiiboToCollected();
    expect(propsMock.addToCollected).toHaveBeenCalledWith(propsMock);
  })

  describe('mapStateToProps', () => {
    it('should return an object with wishlist and collected arrays', () => {
      const expected = { wishlist: [{ id: 1 }], collected: [{ id: 2 }] };
      const mockState = { wishlist: [{ id: 1 }], collected: [{ id: 2 }], extraKey: 'extra' }
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    })
  });

  describe('mapDispatchToProps', () => {
    const dispatchMock = jest.fn();
    const props = mapDispatchToProps(dispatchMock);
    it('should dispatch addToWishlist with an amiibo object as a param', () => {
      const expected = addToWishlist({ id: 1 });
      props.addToWishlist({ id: 1 });
      expect(dispatchMock).toHaveBeenCalledWith(expected);
    });
    it('should dispatch addToCollected with an amiibo object as a param', () => {
      const expected = addToCollected({ id: 2 });
      props.addToCollected({ id: 2 });
      expect(dispatchMock).toHaveBeenCalledWith(expected);
    });
  });
});

