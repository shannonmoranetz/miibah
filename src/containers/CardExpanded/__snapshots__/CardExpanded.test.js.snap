// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardExpanded should match the snapshot 1`] = `
<div
  className="CardExpanded"
>
  <div
    className="inner-card"
  >
    <Link
      className="exit-button"
      replace={false}
      to="/"
    >
      x
    </Link>
    <div
      className="card-image-container"
    >
      <img
        alt="pikachu amiibo"
        className="image-expanded"
        src="pikachu.img"
      />
    </div>
    <div
      className="card-info"
    >
      <h2
        className="card-name"
      >
        pikachu
      </h2>
      <p
        className="card-amiibo-series"
      >
         amiibo series: 
        pokemon
      </p>
      <p
        className="card-game-series"
      >
        game series: 
        pokemon
      </p>
      <p
        className="card-date"
      >
        release date: 
      </p>
    </div>
    <div
      className="expanded-buttons"
    >
      <button
        className="wish-button"
        onClick={[Function]}
      >
        wish
      </button>
      <button
        className="collect-button"
        onClick={[Function]}
      >
        collect
      </button>
    </div>
  </div>
</div>
`;
